_source/_posts/2022-07-21-jhipster-ci-cd.md

```diff
@@ -0,0 +1,933 @@
+
+{% img blog/jhipster-ci-cd/sp-bind-artifact.png alt:"Spinnaker deploy manifest configuration binding artifact" width:"700" %}{: .center-image }
+
+Also add a stage for the `gateway-deployment.yml`, binding the `gateway` Docker image. Finally, add stages for `store-service.yml` and `gateway-service.yml`. Notice Docker images have to be bound only for the deployment manifests, where the images are referenced.
+The complete pipeline must look like this:
+
+{% img blog/jhipster-ci-cd/sp-jhipster-pipeline.png alt:"Spinnaker pipeline for jhipster microservices" width:"800" %}{: .center-image }
+
+Test the pipeline manually once. Go to **Pipelines** and select **Start Manual Execution** for the _store-cd_ pipeline. Set the following options:
+
+- Trigger: leave **docker-account: your-dockerhub-username/store**
+- Type: **Tag**
+- Tag: **latest**
+
+Successful execution will show all stages in a green state:
+
+{% img blog/jhipster-ci-cd/sp-pipeline-success.png alt:"Spinnaker pipeline successful execution" width:"800" %}{: .center-image }
+
+
+Get the external IP of the gateway:
+
+```shell
+kubectl get svc gateway -n demo
+```
+
+Update the redirect URIs in Okta to allow the gateway address as a valid redirect. Run `okta login`, open the returned URL in your browser, and sign in to the Okta Admin Console. Go to the **Applications** section, find your application, edit, and add:
+- Sign-in redirect URIs: `http://<external-ip>:8080/login/oauth2/code/oidc`
+- Sign-out redirect URIs: `http://<external-ip>:8080`
+
+Navigate to `http://<external-ip>:8080` and sign in to the application with your Okta credentials. Then try creating a `Product`:
+
+{% img blog/jhipster-ci-cd/app-create-product.png alt:"Create a product in the store application" width:"800" %}{: .center-image }
+
+### Trigger the CI-CD pipeline with a GitHub push
+
+For testing the workflow, make a code change in the gateway. Edit `src/main/webapp/content/scss/_bootstrap-variables.scss` and update the following variable:
+
+```scss
+$body-bg: steelblue;
+```
+Also, the pipeline under test will only trigger if a new image tag is detected. So edit `.circleci/confg.yml` and update the image name:
+
+```yml
+IMAGE_NAME: indiepopart/gateway:v1
+```
+
+Commit and push the change to the `main` branch, and watch the CircleCI CI pipeline triggers. After the new gateway image is pushed to DockerHub, watch the Spinnaker CD pipeline trigger and deploy the updated gateway. On the left menu choose **Clusters**, and verify the active gateway deployment now has a _V002_. If you click over the **V002** box, you can also verify the image tag that was deployed.
+
+{% img blog/jhipster-ci-cd/sp-version-2.png alt:"Spinnaker deployment version 2" width:"800" %}{: .center-image }
+
```

Comment:
I think it'd be good to add a section here that explains how to delete the clusters you created earlier so you're not charged after doing this tutorial.

Reply:
Added a comment.